signer/src/wsts_state_machine.rs

```diff
@@ -465,41 +484,109 @@ impl SignerStateMachine {
             return Err(error::Error::InvalidConfiguration);
         };
 
-        let state_machine = WstsSigner::new(
+        let inner = WstsSigner::new(
             threshold,
             dkg_threshold,
             num_parties,
             num_keys,
             id,
             key_ids,
-            signer_private_key.into(),
+            private_key.into(),
             public_keys,
-            &mut rng,
+            &mut OsRng,
         )
         .map_err(Error::Wsts)?;
 
-        Ok(Self(state_machine))
+        Ok(Self { inner, started_at, private_key })
+    }
+
+    /// Create a random number generator seeded with the given bitcoin
+    /// block reference and a private key.
+    fn create_rng(started_at: &BitcoinBlockRef, private_key: PrivateKey) -> StdRng {
+        let seed_bytes: [u8; 32] = sha2::Sha256::new_with_prefix("DKG_RNG")
+            .chain_update(started_at.block_hash.into_bytes())
+            .chain_update(started_at.block_height.to_le_bytes())
+            .chain_update(private_key.to_bytes())
+            .finalize()
+            .into();
+
+        StdRng::from_seed(seed_bytes)
```

Comment:
Should we opt to use an explicit RNG here (e.g. from `rand_chacha`) instead of the opaque `StdRng`? IIUC the underlying algorithm for `StdRng` isn't guaranteed between versions.

Reply:
Resolved in [1102863](https://github.com/stacks-sbtc/sbtc/pull/1746/commits/11028635ed1cfdf0011de17f4ea97ed0b97c8d64), closing